### PR TITLE
Add a --fuzz option to make-both-single-timing-files.py 

### PR DESCRIPTION
**Pull request by JasonGross**
Passing `--fuzz=N` allows differences in character locations of up to
`N` characters when matching lines in per-line timing diffs.

See also the discussion at https://github.com/coq/coq/pull/11076#pullrequestreview-324791139

**Kind:** enhancement / infrastructure.

- [x] Added / updated test-suite
- [x] Corresponding documentation was added / updated (including any warning and error messages added / removed / modified).
- [x] Entry added in the changelog (see https://github.com/coq/coq/tree/master/doc/changelog#unreleased-changelog for details).

### CODE REVIEW

File: doc/changelog/08-tools/11230-timing-fuzz.rst [hunk 1]
```diff
@@ -0,0 +1,15 @@
+- **Added:**
+  The ``make-both-single-timing-files.py`` script now accepts a
+  ``--fuzz=N`` parameter on the command line which determines how many
+  characters two lines may be offset in he "before" and "after" timing
```
Comment (Zimmi48) [inline doc/changelog/08-tools/11230-timing-fuzz.rst]:
```suggestion
  characters two lines may be offset in the "before" and "after" timing
```
?

File: doc/changelog/08-tools/11230-timing-fuzz.rst [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  script via the ``print-pretty-single-time-diff`` target in a
+  ``Makefile`` made by ``coq_makefile``, you can set this argument by
+  passing ``TIMING_FUZZ=N`` to ``make``.  (`#11230
+  <https://github.com/coq/coq/pull/11230>`_, by Jason Gross).
+
+- **Added:**
+  Coq's build system now supports both ``TIMING_FUZZ`` and
+  ``TIMING_SORT_BY`` just like a ``Makefile`` made by
+  ``coq_makefile``.  (`#11230
+  <https://github.com/coq/coq/pull/11230>`_, by Jason Gross).
```
Comment (Zimmi48) [inline doc/changelog/08-tools/11230-timing-fuzz.rst]: It looks to me like this is the only place where this second change is documented. Can you add some documentation for this in `INSTALL` for instance?

BTW, this looks like one more thing where we're going to have a regression (or need to port) when we switch to Dune.

Comment (JasonGross) [reply doc/changelog/08-tools/11230-timing-fuzz.rst]: I've added a bit of documentation in `INSTALL`.

Indeed, we should port this to Dune.  I think we need a Dune expert (@ejgallego?) to implement support for `TIMED=1` (per-file timing on stdout by prefixing each command with `/usr/bin/env time -f $* (real: %e, user: %U, sys: %S, mem: %M ko)` on Linux and `gtime -f $* (real: %e, user: %U, sys: %S, mem: %M ko)` if `time -f` doesn't work) and `TIMING=1` (per-line times emitted for each file by passing `-time` to `coqc` and redirecting stdout to a `.timing` file for each `.v` file).  I can probably muddle through figuring out how to make the other targets based on those two, but I imagine that me and @ejgallego could figure it out much more quickly working in concert.

**Review by Zimmi48 — APPROVED** [review]

**Comment by JasonGross** [comment]
Superseded by #11302

**Pull request closed without merging**